### DISCUSSION TITLE
[Возможно ли?] Лайк песни из "Моя волна" > Spotify.

### DISCUSSION THREAD

**Discussion by nik1071** [Chimildic/gooex#2] [Беседка]
Привет всем. Спасибо большое за труды.
 Слушаю допустим мою волну в Яндексе, лайкнул трек. И этот трек сразу отправлялся бы в мой Spotify плейлист. Триггер на каждую минуту если поставить. Возможно ли такое осуществить через gooex?

**Comment by Bedmit** [comment]
В соседней теме выкладывал [свой вариант](https://github.com/Chimildic/gooex/discussions/1#discussioncomment-2610811) экспорта лайков из Spotify в ЯМ и импорта лайков из ЯМ в Spotify. Второе, как я понимаю, как раз то что и нужно. Источник лайков в ЯМ не имеет значения - хоть из волны, хоть из какого угодно плейлиста. Триггер раз в минуту это конечно перебор. Зачем так часто? Раз в сутки мне кажется более чем достаточно.

**Reply by nik1071** [reply]
Просто я слушаю **дома** Мою волну часто, и оттуда треки довольно интересные для себя извлекаю. А библиотека у меня в Spotify и как только нужно уйти куда-то, приходится переносить лайки. Раз в день это крайне редко. Сама библиотека лайков в YM около 3к треков, каждый раз переносить весь плейлист часто, то YM будет блочить наверно(за спам API). А руками переносить треки неудобно(надоело, хочется автоматизировать). Вот и поинтересовался: есть ли возможность синхронизировать не весь плейлист, а лайки последние.

PS ваш скрипт хорошо подошел, спасибо. Просто поставил триггер на синхронизацию последнего трека в лайках YM каждую минуту. Посмотрим как проработает день.

**Reply by Bedmit** [reply]
О-о-очень не советую на "раз в минуту". Это и есть огромный шанс на достижение лимита запросов. У Вас просто Goofy перестанет работать.

В моем скрипте проверяются не все лайки, а только последние 50 (можно изменить в третей строчке). Для меня это вполне достаточно, т..к. даже с моей любовью к лайкам, поставить их больше 50 в день просто не реально. 
Если все же планируется более частый запуск скрипта, это значение нужно уменьшить до разумного. Например 15 треков раз в час (это практически сценарий, когда лайкается все подряд). Ну или совсем уж экстремальный, чаще которого категорически не стоит, 5 треков раз в 15 минут.

Просто неужели возможен сценарий использования, когда Вы каждую минуту куда-то уходите и необходимо переключать сервисы, и прямо обязательно нужно тут же послушать то, что только минуту назад лайкнули?

**Reply by nik1071** [reply]
Я ставил 1 лайк в минуту триггер. Убедили, поставил 5 лайков раз в 15 минут 😀
